src/main/java/com/redhat/cloud/notifications/GwResource.java

```diff
@@ -67,7 +70,12 @@ public Response forward(@NotNull @Valid RestAction ra) {
         Action.Builder builder = Action.newBuilder();
         LocalDateTime parsedTime = LocalDateTime.parse(ra.timestamp, DateTimeFormatter.ISO_LOCAL_DATE_TIME);
         builder.setTimestamp(parsedTime);
-        builder.setPayload(ra.payload);
+        builder.setContext(new HashMap());
+        List<Event> eventList = new ArrayList<>(1);
+        Metadata meta = new com.redhat.cloud.notifications.ingress.Metadata();
+        Event event = new Event(meta, ra.getPayload());
+        eventList.add(event);
```

Comment:
Since we don't need the underlying `Array`, you could use `Collections.singletonList(event)` here.